### PR TITLE
Add New Puzzles components;

### PR DESCRIPTION
**Pull request by kirilknysh**
- Fix navigation (see `useHistory` hook)
- Add `YOLO` mode `on`
- Add initial "new CSS puzzle" component

https://user-images.githubusercontent.com/3236551/131232145-9d947755-f523-44b1-9d8e-bf50a7eff202.mov

### CODE REVIEW

File: src/hooks/useHistory.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+export default function useHistory() {
+  const [history] = useState(() => {
+    const hashHistory = createHashHistory();
+    const hashHistoryListen = hashHistory.listen.bind(hashHistory);
+
+    // hack to fix interfaces mismatch between `history` and `preact-router`
+    // HashHistory calls `listen` callback with an object `{ action, location }`
+    // but preact-router expects `location` right away
+    hashHistory.listen = (callback) => {
+			return hashHistoryListen(({ location }) => {
```
Comment (DaQuirm) [inline src/hooks/useHistory.js:13]: looks like tab indentation 😮

Comment (kirilknysh) [reply src/hooks/useHistory.js:13]: damn... will fix in the next PR (almost there!)

File: src/components/puzzles/NewCss.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      : <span>{'</'}<span>{line.tagName}</span>{'>'}</span>
+    }
+  </div>
+}
+
+function nodesToLines(nodes, indent = 0) {
+  const lines = [];
+
+  nodes.forEach((node, index) => {
+    const qdId = `${indent * 50 + index  }`; // we don't expect more than 50 sibling elements
```
Comment (DaQuirm) [inline src/components/puzzles/NewCss.js:108]: Can we use a simple counter here?

Comment (kirilknysh) [reply src/components/puzzles/NewCss.js:108]: hm, technically... Ideally, a random number would work better. Let me use just a counter (have too many changes already, next PR?)

**Review by DaQuirm — APPROVED** [review]
very good job! 👏

**Pull request merged**